packages/@lwc/engine-core/src/framework/profiler.ts

```diff
@@ -66,27 +66,62 @@ const start = !isUserTimingSupported
 
 const end = !isUserTimingSupported
     ? noop
-    : (measureName: string, markName: string) => {
-          performance.measure(measureName, markName);
+    : (
+          measureName: string,
+          markName: string,
+          devtools?: {
+              color?:
+                  | 'primary'
+                  | 'primary-light'
+                  | 'primary-dark'
+                  | 'secondary'
+                  | 'secondary-light'
+                  | 'secondary-dark'
+                  | 'tertiary'
+                  | 'tertiary-light'
+                  | 'tertiary-dark'
+                  | 'error';
+              properties?: [string, string][];
+          }
+      ) => {
+          performance.measure(measureName, {
+              start: markName,
+              detail: {
+                  devtools: {
+                      dataType: 'track-entry',
+                      track: '⚡️ Lightning Web Components',
+                      ...devtools,
+                  },
+              },
+          });
 
           // Clear the created marks and measure to avoid filling the performance entries buffer.
           // Note: Even if the entries get deleted, existing PerformanceObservers preserve a copy of those entries.
           performance.clearMarks(markName);
           performance.clearMeasures(measureName);
       };
 
-function getOperationName(opId: OperationId): string {
+function getOperationName<T extends OperationId = OperationId>(opId: T) {
     return operationIdNameMapping[opId];
 }
 
-function getMeasureName(opId: OperationId, vm: VM): string {
-    return `${getComponentTag(vm)} - ${getOperationName(opId)}`;
+function getMeasureName<T extends OperationId = OperationId>(opId: T, vm: VM) {
+    return `${getComponentTag(vm)} - ${getOperationName(opId)}` as const;
 }
 
-function getMarkName(opId: OperationId, vm: VM): string {
+function getMarkName<T extends OperationId = OperationId>(opId: T, vm: VM) {
     // Adding the VM idx to the mark name creates a unique mark name component instance. This is necessary to produce
     // the right measures for components that are recursive.
-    return `${getMeasureName(opId, vm)} - ${vm.idx}`;
+    return `${getMeasureName(opId, vm)} - ${vm.idx}` as const;
+}
+
+function getProperties(vm: VM<any, any>): [string, string][] {
+    return [
+        ['tagName', vm.tagName],
+        ['idx', String(vm.idx)],
+        ['renderMode', vm.renderMode === RenderMode.Light ? 'light' : 'shadow'],
+        ['shadowMode', vm.shadowMode === ShadowMode.Native ? 'native' : 'synthetic'],
```

Comment:
```suggestion
    return [
        ['Tag Name', vm.tagName],
        ['Component ID', String(vm.idx)],
        ['Render Mode', vm.renderMode === RenderMode.Light ? 'light DOM' : 'shadow DOM'],
        ['Shadow Mode', vm.shadowMode === ShadowMode.Native ? 'native' : 'synthetic'],
    ];
```

This will look a little nicer in the DevTools given the capitalization they use.

![Screenshot 2024-09-09 at 1 28 30 PM](https://github.com/user-attachments/assets/ad1ee62c-09bc-4ae1-b738-6663c2ce2d0e)